pkg/cloud/services/eks/nodegroup.go

```diff
@@ -475,8 +511,22 @@ func (s *NodegroupService) reconcileNodegroup() error {
 	}
 
 	if eksClusterName, eksNodegroupName := s.scope.KubernetesClusterName(), s.scope.NodegroupName(); ng == nil {
+		conditions.MarkFalse(
+			s.scope.ManagedMachinePool,
+			expinfrav1.EKSNodegroupReadyCondition,
+			expinfrav1.EKSNodegroupCreatingReason,
+			clusterv1.ConditionSeverityInfo,
+			"",
+		)
 		ng, err = s.createNodegroup()
 		if err != nil {
+			conditions.MarkFalse(
+				s.scope.ManagedMachinePool,
+				expinfrav1.EKSNodegroupReadyCondition,
+				expinfrav1.EKSNodegroupFailedToCreateReason,
+				clusterv1.ConditionSeverityError,
+				"",
```

Comment:
Also, here it would be good to set the error message.

Reply:
Updated